pkg/core/blockchain.go

```diff
@@ -1837,7 +1837,7 @@ func (bc *Blockchain) InitVerificationVM(v *vm.VM, getContract func(util.Uint160
 		if err != nil {
 			return fmt.Errorf("%w: %v", ErrInvalidVerification, err)
 		}
-		v.LoadScriptWithFlags(witness.VerificationScript, callflag.ReadOnly)
+		v.LoadScriptWithHash(witness.VerificationScript, hash, callflag.ReadOnly)
```

Comment:
Are you sure that we can safely use it for non-contract witness checks? I thought that `LoadScriptWithHash` is more for contract-based witnesses.

Also, I can't find where the `LoadScriptWithFlags` performs witness hash recalculation (it just don't set the `ctx.scriptHash` at all).

Reply:
> Are you sure that we can safely use it for non-contract witness checks? 

Yes. We know the hash exactly (it's checked some lines above) so we can use `LoadScriptWithHash`.

> it just doesn't set the ctx.scriptHash at all

Yes. But some lines below we load invocation script with `v.LoadScript` which does `ctx.callingScriptHash = v.GetCurrentScriptHash()` and that's where it's being calculated.